packages/cozy-stack-client/src/getIconURL.js

```diff
@@ -99,13 +98,16 @@ const _getIconURL = async (stackClient, opts) => {
     }
     icon = new Blob([icon], { type: mimeTypes[ext] })
   }
-
   return URL.createObjectURL(icon)
 }
 
+/**
+ * We need to catch the error and return a special object in
+ * order to be able to delete the memoization if needed
+ */
+
 const getIconURL = function() {
   return _getIconURL.apply(this, arguments).catch(e => {
-    console.warn(e)
     return ''
   })
```

Comment:
I htink we can reemove the catch, this way we can 

- remove the comment above
- remove the special case for '' in memoize

Reply:
In fact we can't without rewriting the method. 

If we don't catch here, we'll not hit the memoize function and will not hit the cache if needed :/. 

if you have any idea to avoid that, I'll be happy to remove that catch and this ugly return

Reply:
If we really need to return a value here you could return an object of a specific class, or a Symbol.

Like : 
```
class ErrorReturned extends String { }

const getIconURL = function() {
  return _getIconURL.apply(this, arguments).catch(e => {
    return new ErrorReturned
  })
}
```

Then you could check if the return is an instance of ErrorReturned and not only equal to ''. It should not modify in any way the rest of your code

Reply:
And I need to return '' https://github.com/cozy/cozy-client/pull/515/files#diff-e22916779b00265fe0a667cc73319814R44  to not create a breaking change. 

Will give a shot later